### PR TITLE
Update ingress to networking.k8s.io/v1beta1

### LINKED ISSUE #22090: Upgrade Ingress controller to support networking.k8s.io/v1beta1
For https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190125-ingress-api-group.md

### PR DESCRIPTION
**Pull request by howardjohn**
This is supported on 1.14+, the old one is dead in 1.18. In Istio 1.6 we
drop support for 1.13 so its fine to remove the old one.

Fixes https://github.com/istio/istio/issues/22090

### CODE REVIEW

**Review by elfinhe — APPROVED** [review]

**Review by esnible — APPROVED** [review]

File: manifests/base/files/gen-istio-cluster.yaml [hunk 1]
```diff
@@ -5911,10 +5911,10 @@ rules:
 - apiGroups: ["apiextensions.k8s.io"]
   resources: ["customresourcedefinitions"]
   verbs: ["get", "watch", "list"]
-- apiGroups: ["extensions"]
+- apiGroups: ["extensions", "networking.k8s.io"]
```
Comment (hzxuzhonghu) [inline manifests/base/files/gen-istio-cluster.yaml]: why still keep the extensions group?

File: istioctl/cmd/convert_ingress.go [hunk 2]
```diff
@@ -97,7 +97,8 @@ func readConfigs(readers []io.Reader) ([]model.Config, []*v1beta1.Ingress, error
 		recognized := 0
 		for _, nonIstio := range kinds {
 			if nonIstio.Kind == "Ingress" &&
-				nonIstio.APIVersion == "extensions/v1beta1" {
+				(nonIstio.APIVersion == "extensions/v1beta1" ||
```
Comment (hzxuzhonghu) [inline istioctl/cmd/convert_ingress.go]: Still keep support for extensions in cli?

Comment (howardjohn) [reply istioctl/cmd/convert_ingress.go]: it's basically free so I don't see why we shouldn't, any reasons I am missing?

File: manifests/base/files/gen-istio-cluster.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/base/files/gen-istio-cluster.yaml]: good point, I can drop this

Comment (hzxuzhonghu) [reply manifests/base/files/gen-istio-cluster.yaml]: You should it is useless now

File: istioctl/cmd/convert_ingress.go (hunk 2 above)
Comment (hzxuzhonghu) [reply istioctl/cmd/convert_ingress.go]: Can come up with one point: to keep consistent

File: manifests/base/files/gen-istio-cluster.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/base/files/gen-istio-cluster.yaml]: done

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**